src/conn_poll.c

```diff
@@ -31,33 +40,196 @@ typedef struct registry_impl {
 
 typedef enum conn_state { CONN_STATE_NEW = 0, CONN_STATE_READY, CONN_STATE_FINISHED } conn_state_t;
 
+typedef enum poll_entry_type {
+	POLL_ENTRY_INTERRUPT = 0,
+	POLL_ENTRY_UDP,
+	POLL_ENTRY_TCP_LISTENER,
+	POLL_ENTRY_TCP_CONN,
+} poll_entry_type_t;
+
+typedef struct tcp_connection {
+	socket_t sock;
+	addr_record_t remote;
+	bool closed;
+	bool connecting;
+	uint16_t frame_len;
+	size_t header_bytes;
+	size_t payload_len;
+	uint8_t *send_buffer;
+	size_t send_size;
+	size_t send_offset;
+	size_t send_capacity;
+	uint8_t recv_buffer[65535];
```

Comment:
Please use the existing `tcp_ice_write_context_t` and `tcp_ice_read_context_t`. By the way, this allocates 66KB for every TCP connection, even failed ones, which is a lot.